sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/arm/implementation/ManagerBase.java

```diff
@@ -14,43 +16,52 @@
 
     private ResourceManager resourceManager;
     private final String subscriptionId;
-    protected final RestClient restClient;
+    private final AzureEnvironment environment;
+    protected final HttpPipeline httpPipeline;
     private SdkContext sdkContext;
 
-    protected ManagerBase(RestClient restClient, String subscriptionId, SdkContext sdkContext) {
-        this.restClient = restClient;
-        if (restClient != null) {
-            this.resourceManager = ResourceManager.authenticate(restClient)
+    protected ManagerBase(HttpPipeline httpPipeline, AzureProfile profile, SdkContext sdkContext) {
+        this.httpPipeline = httpPipeline;
+        if (httpPipeline != null) {
+            this.resourceManager = ResourceManager.authenticate(httpPipeline, profile)
                     .withSdkContext(sdkContext)
-                    .withSubscription(subscriptionId);
+                    .withDefaultSubscription();
```

Comment:
why resourcemanager use default subs, not profile.subsid?

Reply:
This is the step to create instance of `ResourceManager` rather than instance of `ResourceManager.Authenticated`. It will take subscription id from profile.